File: 0222-count-complete-tree-nodes/0222-count-complete-tree-nodes.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def countNodes(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        if not root: #if root is None return 0 since there is no tree
            return 0
        
        if not root.left and not root.right: #if there are no child nodes return 1 as it is only the root
            return 1

        left_count = self.countNodes(root.left)
        right_count = self.countNodes(root.right)

        return left_count + right_count + 1

File: 0222-count-complete-tree-nodes/0222-count-complete-tree-nodes.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    public int CountNodes(TreeNode root) {
        if (root == null)
        {
            return 0;
        }
        if(root.left == null && root.right == null)
        {
            return 1;
        }
        int left = this.CountNodes(root.left);
        int right = this.CountNodes(root.right);
        
        return left + right + 1;
    }
}